src/main/java/com/silzila/querybuilder/QueryComposer.java

```diff
@@ -49,19 +49,19 @@ public String composeQuery(Query req, DatasetDTO ds, String vendorName) throws B
 
         if (vendorName.equals("postgresql") || vendorName.equals("redshift")) {
             // System.out.println("------ inside postges block");
-            qMap = SelectClausePostgres.buildSelectClause(req);
+            qMap = SelectClausePostgres.buildSelectClause(req, vendorName);
         } else if (vendorName.equals("mysql") || vendorName.equals("duckdb")) {
             // System.out.println("------ inside mysql block");
-            qMap = SelectClauseMysql.buildSelectClause(req);
+            qMap = SelectClauseMysql.buildSelectClause(req, vendorName);
         } else if (vendorName.equals("sqlserver")) {
             // System.out.println("------ inside sql server block");
-            qMap = SelectClauseSqlserver.buildSelectClause(req);
+            qMap = SelectClauseSqlserver.buildSelectClause(req, vendorName);
         } else if (vendorName.equals("bigquery")) {
             // System.out.println("------ inside Big Query block");
-            qMap = SelectClauseBigquery.buildSelectClause(req);
+            qMap = SelectClauseBigquery.buildSelectClause(req, vendorName);
         } else if (vendorName.equals("databricks")) {
             // System.out.println("------ inside databricks block");
-            qMap = SelectClauseDatabricks.buildSelectClause(req);
+            qMap = SelectClauseDatabricks.buildSelectClause(req, vendorName);
```

Comment:
Why are you passing vendorName?
under SelectClausePostgres, we know for sure that is it for Postgres.
And same goes for other Select clause database files.